### PR TITLE
Remove mutable containers functionality.

### PR DESCRIPTION
**Pull request by richardjrossiii**
Horrible, horrible feature, removing this will bring us up to parity with the iOS and Android repos.

### CODE REVIEW

File: Parse/ParseUser.cs [hunk 1]
```diff
@@ -558,6 +558,7 @@ public class ParseUser : ParseObject {
           authData = AuthData = new Dictionary<string, IDictionary<string, object>>();
         }
         authData[authType] = data;
+        AuthData = authData;
```
Comment (richardjrossiii) [inline Parse/ParseUser.cs]: This is needed, as we were previously relying on the dictionary magically being updated.

File: Parse/Parse.csproj [hunk 2]
```diff
@@ -142,9 +141,5 @@
   <Target Name="AfterBuild">
   </Target>
   -->
-  <ItemGroup>
-    <Folder Include="Internal\Config\" />
-    <Folder Include="Internal\Push\State\" />
-    <Folder Include="Internal\Push\Coder\" />
-  </ItemGroup>
+  <ItemGroup />
```
Comment (hallucinogen) [inline Parse/Parse.csproj]: <3 we can kill this line too

File: Parse/ParseUser.cs (hunk 1 above)
Comment (hallucinogen) [inline Parse/ParseUser.cs]: nice catch

**Comment by hallucinogen** [comment]
Kill it with fire.

Make sure IntegrationTests run and green since we haven't ported lots of important ParseObject tests from Integration to Unit.

File: Parse/Parse.csproj (hunk 2 above)
Comment (richardjrossiii) [inline Parse/Parse.csproj]: I think it was actually generated by MonoDevelop. I'll try removing it and see how much breaks.

**Comment by richardjrossiii** [comment]
All integration tests pass! Woo! Merging as soon as travis passes

**Pull request merged**